src/pcre2_compile.c

```diff
@@ -5388,14 +5426,45 @@ if ((options & PCRE2_CASELESS) != 0)
   if ((options & (PCRE2_UTF|PCRE2_UCP)) != 0)
     {
     int rc;
-    uint32_t oc, od;
+    uint32_t oc, od, skip_start;
+    const uint32_t *skip_range;
 
     options &= ~PCRE2_CASELESS;   /* Remove for recursive calls */
     c = start;
+    skip_range = get_nocase_range(c);
+    skip_start = skip_range[0];
+    if (c > skip_start)
+      {
+      c = skip_range[1];
+      skip_range += 2;
+      skip_start = skip_range[0];
+      }
 
     while ((rc = get_othercase_range(&c, end, &oc, &od,
              (xoptions & PCRE2_EXTRA_CASELESS_RESTRICT) != 0)) >= 0)
       {
+      if (c > skip_start)
```

Comment:
I thought get_othercase_range returns the `c` after the character with othercase, but sometimes it moves even further. I am not fully understand this.